lib/matplotlib/image.py

```diff
@@ -361,6 +362,13 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                     a_min, a_max = np.int32(0), np.int32(1)
                 if inp_dtype.kind == 'f':
                     scaled_dtype = A.dtype
+                    # Cast to float64
+                    if not (A.dtype == np.float32 or A.dtype == np.float16):
```

Comment:
`if A.dtype not in (np.float32, np.float16):`

Reply:
Done.